csi/controllerserver.py

```diff
@@ -197,9 +197,60 @@ def CreateVolume(self, request, context):
                     vol = create_virtblock_volume(
                         mntdir, request.name, pvsize)
                 else:
+                    use_gluster_quota = False
+                    if os.path.isfile("/etc/secret-volume/ssh-privatekey") and "SECRET_GLUSTERQUOTA_SSH_USERNAME" in os.environ:
+                        use_gluster_quota = True
+                    secret_private_key = "/etc/secret-volume/ssh-privatekey"
+                    secret_username = os.environ.get('SECRET_GLUSTERQUOTA_SSH_USERNAME', None)
+                    hostname = filters.get("gluster_hosts", None)
+                    gluster_vol_name = filters.get("gluster_volname", None)
                     vol = create_subdir_volume(
-                        mntdir, request.name, pvsize)
-
+                        mntdir, request.name, pvsize, use_gluster_quota)
+                    quota_size = pvsize
+                    quota_path = vol.volpath
+                    if use_gluster_quota == False:
+                        logging.debug(logf("Set Quota in the native way"))
+                    else:
+                        logging.debug(logf("Set Quota using gluster directory Quota"))
+                        quota_cmd = [
+                            "ssh",
+                            "-oStrictHostKeyChecking=no",
+                            "-i",
+                            "%s" % secret_private_key, 
+                            "%s@%s" % (secret_username, hostname),
+                            "sudo",
+                            "gluster",
+                            "volume",
+                            "quota",
+                            "%s" % gluster_vol_name,
+                            "limit-usage",
+                            "/%s" % quota_path,
+                            "%s" % quota_size
+                        ]
+                        
+                        quota_deem_cmd = [
```

Comment:
I personally think, 'deem-statfs' option setting should be done just 1 time, so, good to move it to 'mount_glusterfs()' external part.